Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.cpp

```diff
@@ -383,12 +385,107 @@ FloatRect LegacyRenderSVGShape::calculateStrokeBoundingBox() const
     return adjustStrokeBoundingBoxForMarkersAndZeroLengthLinecaps(RepaintRectCalculation::Accurate, strokeBoundingBox);
 }
 
+FloatRect LegacyRenderSVGShape::calculateApproximateScalingStrokeBoundingBox(FloatRect fillBoundingBox) const
+{
+    // Implementation of
+    // https://drafts.fxtf.org/css-masking/#compute-stroke-bounding-box
+    // except that we ignore whether the stroke is none.
+
+    FloatRect strokeBoundingBox = fillBoundingBox;
+    const float strokeWidth = this->strokeWidth();
+    if (strokeWidth <= 0)
+        return strokeBoundingBox;
+
+    float delta = strokeWidth / 2;
+    switch (m_shapeType) {
+    case ShapeType::Empty: {
+        // // Spec: "A negative value is illegal. A value of zero disables rendering of the element."
+        return strokeBoundingBox;
+    }
+    case ShapeType::Ellipse:
+    case ShapeType::Circle: {
+        break;
+    }
+    case ShapeType::Rectangle:
+    case ShapeType::RoundedRectangle: {
+#if USE(CG)
+        // CoreGraphics can inflate the stroke by 1px when drawing a rectangle with antialiasing disabled at non-integer coordinates, we need to compensate.
+        if (style().svgStyle().shapeRendering() == ShapeRendering::CrispEdges)
+            delta += 1;
+#endif
+        break;
+    }
+    case ShapeType::Path:
+    case ShapeType::Line: {
+        auto& style = this->style();
+        if (m_shapeType == ShapeType::Path && style.joinStyle() == LineJoin::Miter) {
+            const float miter = style.strokeMiterLimit();
+            if (miter < sqrtOfTwoDouble && style.capStyle() == LineCap::Square)
+                delta *= sqrtOfTwoDouble;
+            else
+                delta *= std::max(miter, 1.0f);
+        } else if (style.capStyle() == LineCap::Square)
+            delta *= sqrtOfTwoDouble;
+        break;
+    }
+    }
+
+    strokeBoundingBox.inflate(delta);
+    return strokeBoundingBox;
+}
+
+FloatRect LegacyRenderSVGShape::calculateApproximateNonScalingStrokeBoundingBox(FloatRect fillBoundingBox) const
+{
+    ASSERT(hasPath());
+    ASSERT(style().svgStyle().hasStroke());
+    ASSERT(hasNonScalingStroke());
+
+    FloatRect strokeBoundingBox = fillBoundingBox;
+    AffineTransform nonScalingTransform = nonScalingStrokeTransform();
+    if (std::optional<AffineTransform> inverse = nonScalingTransform.inverse()) {
+        Path* usePath = nonScalingStrokePath(m_path.get(), nonScalingTransform);
+        FloatRect strokeBoundingRect = calculateApproximateScalingStrokeBoundingBox(usePath->fastBoundingRect());
```

Comment:
General comment, which I should've mentioned on the previous patches too: prefer to use `auto` / `auto*` / `auto&` unless it's confusing or unclear not to know the exact type or you explicitly want to use a type different from what `auto` would give you. Here I would use `auto` for all of these declarations. (Existing code might not already follow this guideline well.)

Reply:
Thanks! I've changed :D